libs/api.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Class for handling all requests to the server
+ */
+class API {
+    /**
+     * Initializes the api connection
+     * @param {string} host - base URL for all requests (should include https://)
+     */
+    constructor(host) {
+        this.host = host;
+    }
```

Comment:
As this is refactoring it perhaps a bit late. But the _server_ or _host_ are very general terms. Perhaps as this is for our purposes, nice to mention that the host should point to the dataserver where the data should be uploaded